src/main/java/build/buildfarm/server/ExecutionService.java

```diff
@@ -69,12 +69,12 @@ private void withCancellation(
         future,
         new FutureCallback<Void>() {
           boolean isCancelled() {
-            return serverCallStreamObserver.isCancelled() || Context.current().isCancelled();
+            return !serverCallStreamObserver.isCancelled() && !Context.current().isCancelled();
```

Comment:
isn't expression now testing is `isNotCancelled`? maybe change method:  You could change to `isActive`? I'm guessing that static analysis just wanted to avoid `if (!expressions)`?

Reply:
Will revert this one.